client/src/hooks/useLogout.js

```diff
@@ -0,0 +1,30 @@
+// SPDX-FileCopyrightText: 2022 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import userService from '../services/user';
+import useAuth from './useAuth';
+
+// custom hook for logout, used since no individual logout page will be created
+
+const useLogout = () => {
+
+  const { setAuth } = useAuth();
+  
+  const logout = async () => {
+
+    setAuth({});
```

Comment:
Can this be set before the logout() below has been successful? Does this code enable a situation where the logout fails (so the user is still "logged in") but the auth (context) is empty.